src/core/execute.c

```diff
@@ -4167,6 +4167,19 @@ static int exec_child(
                 *exit_status = EXIT_MEMORY;
                 return log_oom();
         }
+        /* The PATH variable is set to the default path in params->environment.
+         * This is why we do not check accum_env and instead check context->environment,
+         * files_env, and pass_env individually. We override PATH if the user does
+         * not specify PATH and the user has specified ExecSearchPath
+         */
+        if (!strv_env_get(context->environment, "PATH") && !strv_env_get(files_env, "PATH") && !strv_env_get(pass_env, "PATH") && context->exec_search_paths) {
+                r = strv_env_assign(&accum_env, "PATH", context->exec_search_paths);
+                if (r < 0) {
+                        *exit_status = EXIT_MEMORY;
+                        return log_oom();
+                }
+        }
```

Comment:
it's cleaner to synthesize an env block from the search path, and then just pass it into strv_env_merge(), and let that do its thing. The override rules should be the ones implemented in strv_env_merge(), and there shouldn't be another level of overriding outside of it.

Reply:
Was under the impression that PATH should be overridden only if it is supplied by the user. Meaning through Environment=, EnvironmentFile=, and PassEnvironment=. If we pass it into strv_env_merge() we do not have a way to differentiate between user supplied and not. ExecContent also has a PATH variable, but that is not supplied by the user as far as I know. Do we want to override in all cases if exec_search_path is supplied?

Reply:
@poettering Any guidance regarding this?

Reply:
strv_env_merge() can merge any number of environment blocks. The later ones in the parameter list will override the maching variables earlier in the list. Hence, just add one more parameter to the function, and put it early in the list, so that it gets overriden by whatever the user specifies.